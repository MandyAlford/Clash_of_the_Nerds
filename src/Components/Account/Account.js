import React from 'react';
import './Account.scss';
import { connect } from 'react-redux';
import AccountResults from '../AccountResults/AccountResults'
import PropTypes from 'prop-types';

export const Account = (props) => {

  const triviaCards = props.saveTrivia.map(trivia => <AccountResults key={Math.random()} result={trivia}/>)
  return (
    <section className='account-page'>
      <h1 className='account-name'>{props.user[0].name}'s Account Page</h1>
      <div>{triviaCards}</div>
    </section>
  )
}

Account.propTypes = {
  saveTrivia: PropTypes.array
}

export const mapStateToProps = state => ({
  user: state.user,
  saveTrivia: state.saveTrivia
});

export default connect(mapStateToProps)(Account)
