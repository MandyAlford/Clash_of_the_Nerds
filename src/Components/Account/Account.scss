.account-page {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100vh;
}

.account-name {
  margin-top: 50px;
  width: 70%;
  margin-left: 15%;
  background-color:  #662324;
  height: 50px;
  color: #FFF;
  font-size: 30px;
}
