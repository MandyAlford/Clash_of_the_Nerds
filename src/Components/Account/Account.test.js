import React from 'react';
import { Account, mapStateToProps } from './Account';
import { shallow } from 'enzyme';
import { mockRandom } from 'jest-mock-random';


describe('Account', () => {
  mockRandom([0.1]);
  let wrapper = shallow(<Account
      saveTrivia={[
        {category: 'Mythology',
        type: 'multiple',
        difficulty: 'easy',
        question: 'The greek god Poseidon was the god of what?',
        correct_answer: 'The Sea',
        incorrect_answers: ['War', 'Sun', 'Fire']
      }]}
      user={[{
        name: 'Jessica Rabbit',
        team: '19',
        level: 'hard'
      }]}
    />)

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {
    it('should be able to update state', () => {
      let mockState= {
        user: {
          name: 'Jessica Rabbit',
          team: '19',
          level: 'hard'
        },
        saveTrivia: [
          {category: 'Mythology',
          type: 'multiple',
          difficulty: 'easy',
          question: 'The greek god Poseidon was the god of what?',
          correct_answer: 'The Sea',
          incorrect_answers: ['War', 'Sun', 'Fire']
        }]
      };
      const expected = {
        user: {
          name: 'Jessica Rabbit',
          team: '19',
          level: 'hard'
        },
        saveTrivia: [
          {category: 'Mythology',
          type: 'multiple',
          difficulty: 'easy',
          question: 'The greek god Poseidon was the god of what?',
          correct_answer: 'The Sea',
          incorrect_answers: ['War', 'Sun', 'Fire']
        }]
      };
      const result = mapStateToProps(mockState)
      expect(result).toEqual(expected)
    })
  })

})
