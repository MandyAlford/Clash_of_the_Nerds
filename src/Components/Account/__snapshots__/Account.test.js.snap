// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account should match snapshot 1`] = `
<section
  className="account-page"
>
  <h1
    className="account-name"
  >
    Jessica Rabbit
    's Account Page
  </h1>
  <div>
    <Connect(TriviaResults)
      result={
        Object {
          "category": "Mythology",
          "correct_answer": "The Sea",
          "difficulty": "easy",
          "incorrect_answers": Array [
            "War",
            "Sun",
            "Fire",
          ],
          "question": "The greek god Poseidon was the god of what?",
          "type": "multiple",
        }
      }
    />
  </div>
</section>
`;
